views/pages/ConfigurationsPage/index.jsx

```diff
@@ -118,25 +120,72 @@ const ConfigurationsPage = () => {
     }
   }
 
+  const handleConfigurationUpload = (e) => {
+    try {
+      const file = e.target.files ? e.target.files[0] : ''
+
+      new Response(file).json().then(async (newConfigurationAttributes) => {
+        newConfigurationAttributes.owner = user.uid
+        newConfigurationAttributes.readers = [user.uid]
+        newConfigurationAttributes.type = 'matrix'
+
+        await api.userConfigurations.create(
+          user.uid,
+          newConfigurationAttributes
+        )
+        await loadAllConfigurations(uid)
+
+        handleNotification('New configuration added')
+      })
```

Comment:
is this how the app worked previously or is this a new implementation?

Reply:
@mikestone14 Not sure exactly about the answer to your question, but Iván and I discussed this and decided that this approach to the old feature should be a client-side file upload that we parse and send through the API. This lets us reuse all the schema validation and security in case we get a malformed or malicious file. Even if the old implementation shipped the file itself to the server, we don't want to do that with the new validation features we've added